### DISCUSSION TITLE
Bounce-like animation on SVG icon inside button in Tailwind?

### DISCUSSION THREAD

**Discussion by deadcoder0904** [tailwindlabs/tailwindcss#4162] [Help] [answered]
How do I get this animation effect using pure Tailwind → https://codepen.io/bekahmcdonald/pen/rrmdwj

I created a Tailwind Play link → https://play.tailwindcss.com/6uHgZwsqGt

Copy-pasting here for brevity:

```html
<div class="flex justify-center items-center h-screen">
  <button class="inline-flex items-center text-white px-4 py-2 text-sm font-medium border border-transparent rounded-md shadow-sm select-none focus:outline-none focus:ring-2 focus:ring-offset-2 hover:shadow-lg transform hover:-translate-y-0.5 transition-all duration-150 group bg-indigo-600 hover:bg-indigo-700 focus:ring-indigo-500">
    <svg class="w-6 h-6 text-white group-hover:animate-bounce" fill="none" viewBox="0 0 24 24" stroke="currentColor">
      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M15 13l-3 3m0 0l-3-3m3 3V8m0 13a9 9 0 110-18 9 9 0 010 18z"></path>
    </svg>
    <span class="ml-2">Download Image</span>
  </button>
</div>
```

> > Note: The `group-hover:animation-bounce` doesn't work on Tailwind Play but works locally so u can use this as a starting point (I have created an issue on the Tailwind Play repo about it → https://github.com/tailwindlabs/play.tailwindcss.com/issues/54)

`animation-bounce` goes too high & keeps bouncing. I want a subtle animation that's not too much in your face.

**Comment by olets** [comment]
Out of the box, `group-hover` is not enabled for `animation`. You'll need to update your config to turn on the `group-hover` variant. See https://tailwindcss.com/docs/animation#variants

**Reply by deadcoder0904** [reply]
@olets but that only resolves the issue for Tailwind Play but anyways it works locally, granted I'm using JIT. Does it mean JIT enables it by default?

Although, my original issue still stands :)

**Reply by deadcoder0904** [reply]
@olets I think my local config worked because of JIT only. I did add `group-hover` to `animation` & it worked.

# tailwind.config.js

```js
module.exports = {
  theme: {},
  variants: {
    extend: {
      animation: ['group-hover'],
    },
  },
  plugins: [],
}
```

**Reply by olets** [reply]
Yes JIT obviates having to select variants.

I see, I misunderstood your question. You are wondering how to put that animation into Tailwind? Use the setup in your selected answer, but move the codepen's entire `keyframes` into your Tailwind `keyframes`.

**Reply by deadcoder0904** [reply]
@olets already got that and posted it as [an answer below](https://github.com/tailwindlabs/tailwindcss/discussions/4162#discussioncomment-652707), thank you :)

**Reply by olets** [reply]
Yep! I meant re: "doesn't look as professional" that you could use the exact same effect as in the codepen

**Comment by deadcoder0904** [comment, marked as answer]
I used a custom animation to get that effect although the animation doesn't look as professional:

# tailwind.config.js

```js
module.exports = {
	purge: ['./src/**/*.{ts,tsx}'],
	darkMode: 'class', // or 'media' or 'class'
	theme: {
		extend: {
			animation: { bounce: 'bounce 0.3s 1' },
			keyframes: {
				bounce: {
					'0%, 100%': {
						transform: 'translateY(-10%)',
						animationTimingFunction: 'ease-in-out',
					},
					'50%': {
						transform: 'translateY(0)',
						animationTimingFunction: 'ease-out',
					},
				},
			},
		},
	},
	variants: {},
}
```
